OneSignalSDK/onesignal/core/src/test/java/com/onesignal/user/internal/migrations/RecoverFromDroppedLoginBugTests.kt

```diff
@@ -31,16 +33,24 @@ class RecoverFromDroppedLoginBugTests : FunSpec({
             )
         every { mockOperationModelStore.loadOperations() } just runs
         every { mockOperationModelStore.list() } returns listOf()
-
-        val recovery = RecoverFromDroppedLoginBug(operationRepo, MockHelper.identityModelStore(), mockConfigModelStore)
-        every { recovery.onOperationRepoLoaded() } just runs
+        val recovery = spyk(RecoverFromDroppedLoginBug(operationRepo, MockHelper.identityModelStore(), mockConfigModelStore))
 
         // When
-        operationRepo.start()
         recovery.start()
+        val waiter = Waiter()
+        operationRepo.addOperationLoadedListener(
+            object : IOperationRepoLoadedListener {
+                override fun onOperationRepoLoaded() {
+                    waiter.wake()
+                }
+            },
+        )
+        operationRepo.start()
```

Comment:
Should we change the order so that the operationRepo starts before the recovery starts, since that is most of the case in real environment?

Reply:
True that it would be more real world, but it also means it could be flaky. This realization was why I created PR #2084 to change the production code.